lib/ansible/modules/cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,267 @@
+options:
+    name:
+        description:
+            - The name of the provider to create.
+        required: true
+        type: str
+    saml_metadata_document:
+        description:
+            - The XML document generated by an identity provider (IdP) that supports SAML 2.0.
+        type: str
+    state:
+        description:
+            - Whether to create or delete identity provider. If 'present' is specified it will attempt to update the identity provider matching the name field.
+        default: present
+        choices: [ "present", "absent" ]
+        type: str
+extends_documentation_fragment:
+    - aws
+    - ec2
+author:
+    - Tony (@axc450)
+    - Aidan Rowe (@aidan-)
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+# It is assumed that their matching environment variables are set.
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+      name: example1
+      # the > below opens an indented block, so no escaping/quoting is needed when in the indentation level under this key
+      saml_metadata_document: >
+          <?xml version="1.0"?>...
+          <md:EntityDescriptor
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+      name: example2
+      saml_metadata_document: "{{ item }}"
+  with_file: /path/to/idp/metdata.xml
+# Removes iam saml identity provider
+- name: remove saml provider
+  iam_saml_federation:
+      name: example3
+      state: absent
+'''
+
+RETURN = '''
+saml_provider:
+    description: Details of the SAML Identity Provider that was created/modified.
+    type: complex
+    returned: present
+    contains:
+        arn:
+            description: The ARN of the identity provider.
+            type: str
+            returned: present
+            sample: "arn:aws:iam::123456789012:saml-provider/my_saml_provider"
+        metadata_document:
+            description: The XML metadata document that includes information about an identity provider.
+            type: str
+            returned: present
+        create_date:
+            description: The date and time when the SAML provider was created in ISO 8601 date-time format.
+            type: str
+            returned: present
+            sample: "2017-02-08T04:36:28+00:00"
+        expire_date:
+            description: The expiration date and time for the SAML provider in ISO 8601 date-time format.
+            type: str
+            returned: present
+            sample: "2017-02-08T04:36:28+00:00"
+'''
+
+try:
+    import boto3
+    import botocore.exceptions
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import AWSRetry, aws_common_argument_spec, get_aws_connection_info, boto3_conn
```

Comment:
`aws_common_argument_spec` is currently unused, `get_aws_connection_info` and `boto3_conn` shouldn't be necessary.